src/handlers/imgix.ts

```diff
@@ -0,0 +1,107 @@
+import ImgixClient from 'imgix-core-js';
+import {
+  IMGIX_DOMAIN as domain,
+  IMGIX_TOKEN as secureURLToken,
+  // @ts-ignore
+} from 'react-native-dotenv';
+import { Source } from 'react-native-fast-image';
+import parse from 'url-parse';
+
+const shouldCreateImgixClient = (): ImgixClient | null => {
+  if (
+    typeof domain === 'string' &&
+    !!domain.length &&
+    typeof secureURLToken === 'string' &&
+    !!secureURLToken.length
+  ) {
+    return new ImgixClient({
+      domain,
+      includeLibraryParam: false,
+      secureURLToken,
+    });
+  }
+  // eslint-disable-next-line no-console
+  console.log(
+    '[Imgix] Image signing disabled. Please ensure you have specified both IMGIX_DOMAIN and IMGIX_TOKEN inside your .env.'
+  );
+  return null;
+};
+
+const staticImgixClient = shouldCreateImgixClient();
+
+// Here we use a static buffer to prevent multiple successive signing attempts
+// for components which may have been unmounted/remounted. This is done to
+// increase performance. It's important to recognize that this cache is
+// **unbounded** and will grow **forever**, so we should probably opt to use
+// an LRU to cap memory. https://yomguithereal.github.io/mnemonist/lru-cache.html
+const staticSignatureCache = {} as {
+  readonly [externalImageUri: string]: string;
+};
+
+const shouldSignUri = (externalImageUri: string): string | undefined => {
+  try {
+    // We'll only attempt to sign if there's an available client. A client
+    // will not exist if the .env hasn't been configured correctly.
+    if (staticImgixClient) {
+      // eslint-disable-next-line no-console
+      __DEV__ && console.log(`[Imgix]: Signed "${externalImageUri}".`);
+      return staticImgixClient.buildURL(externalImageUri, {});
+    }
```

Comment:
if there's no static client we should return the original url, otherwise images won't work for open source contributors

Reply:
That _should_ be okay! If the client is available, it should fall back to the default:

```typescript
const shouldSignUri = (externalImageUri: string): string | undefined => {
  try {
    // We'll only attempt to sign if there's an available client. A client
    // will not exist if the .env hasn't been configured correctly.
    if (staticImgixClient) {
      // eslint-disable-next-line no-console
      __DEV__ && console.log(`[Imgix]: Signed "${externalImageUri}".`);
      return staticImgixClient.buildURL(externalImageUri, {});
    }
  } catch (e) {
    // eslint-disable-next-line no-console
    console.log(`[Imgix]: Failed to sign "${externalImageUri}"!`, e);
    // If something goes wrong, it is not safe to assume the image is valid.
    return undefined;
  }
  return externalImageUri;
};
```